src/cli/cli_tcp.cpp

```diff
@@ -74,13 +98,62 @@ template <> otError TcpExample::Process<Cmd("init")>(Arg aArgs[])
     }
     else
     {
-        if (aArgs[0] == "circular")
+        if (aArgs[0] == "tls")
         {
             mUseCircularSendBuffer = true;
+            mUseTLS                = true;
+
+            // mbedtls_debug_set_threshold(0);
+
+            mbedtls_ctr_drbg_init(&mCtrDrbg);
+            mbedtls_x509_crt_init(&mSrvCert);
+            mbedtls_pk_init(&mPKey);
+
+            mbedtls_ssl_init(&mSSLContext);
+            mbedtls_ssl_config_init(&mSSLConfig);
+            mbedtls_ssl_conf_rng(&mSSLConfig, mbedtls_ctr_drbg_random, &mCtrDrbg);
+            // mbedtls_ssl_conf_dbg(&mSSLConfig, MbedTlsDebugOutput, this);
+            mbedtls_ssl_conf_authmode(&mSSLConfig, MBEDTLS_SSL_VERIFY_NONE);
+            mbedtls_entropy_init(&mEntropy);
+
+            int rv = mbedtls_pk_parse_key(&mPKey, reinterpret_cast<const unsigned char *>(sSrvKey), sSrvKeyLength,
```

Comment:
See https://github.com/Mbed-TLS/mbedtls/blob/development/docs/3.0-migration-guide.md#some-functions-gained-an-rng-parameter

Reply:
```
#if (MBEDTLS_VERSION_NUMBER >= 0x03000000)
#include "crypto/mbedtls.hpp"
            int rv = mbedtls_pk_parse_key(&mPKey, reinterpret_cast<const unsigned char *>(sSrvKey), sSrvKeyLength,
                                          nullptr, 0, Crypto::MbedTls::CryptoSecurePrng, nullptr);
#else
            int rv = mbedtls_pk_parse_key(&mPKey, reinterpret_cast<const unsigned char *>(sSrvKey), sSrvKeyLength,
                                          nullptr, 0);
#endif
```

Reply:
Thanks. I've included this in my next commit.